Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -151,110 +145,69 @@ private static string AssertPluginJarExists(string rootDir)
             Assert.AreEqual(1, files.Length, "Expecting one and only one jar file to be created");
             return files.First();
         }
-
-        private static void AssertExpectedPropertyDefinitionValue(string propertyName, string expected, JarInfo jarInfo)
-        {
-            JarInfo.PropertyDefinition actual = AssertPropertyDefinitionExists(propertyName, jarInfo);
-            Assert.AreEqual(expected, actual.DefaultValue, "Property definition does not have the expected value. Property: {0}", propertyName);
-        }
-
-        private static JarInfo.PropertyDefinition AssertPropertyDefinitionExists(string propertyName, JarInfo jarInfo)
-        {
-            Assert.IsNotNull(jarInfo.Extensions, "Extensions should not be null");
-
-            JarInfo.PropertyDefinition actual = jarInfo.Extensions
-                .OfType<JarInfo.PropertyDefinition>()
-                .FirstOrDefault(pd => string.Equals(pd.Key, propertyName, System.StringComparison.Ordinal));
-
-            Assert.IsNotNull(actual, "Failed to find expected property: {0}", propertyName);
-            AssertPropertyHasValue(actual.DefaultValue, propertyName);
-            return actual;
-        }
-
-        private static JarInfo.RulesDefinition AssertRulesDefinitionExists(JarInfo jarInfo)
+        
+        private static void AssertExpectedPropertyDefinitionValue(string propertyName, string expectedValue, RoslynSdkConfiguration actualConfig)
         {
-            Assert.IsNotNull(jarInfo.Extensions, "Extensions should not be null");
-
-            IEnumerable<JarInfo.RulesDefinition> defns = jarInfo.Extensions.OfType<JarInfo.RulesDefinition>();
+            Assert.IsNotNull(actualConfig.Properties, "Configuration Properties should not be null");
 
-            Assert.AreNotEqual(0, defns.Count(), "RulesDefinition extension does not exist");
-            Assert.AreEqual(1, defns.Count(), "Multiple rules definitions exist");
+            Assert.IsTrue(actualConfig.Properties.ContainsKey(propertyName), "Expected property is not set: {0}", propertyName);
 
-            JarInfo.RulesDefinition defn = defns.Single();
-
-            return defn;
+            Assert.AreEqual(expectedValue, actualConfig.Properties[propertyName], "Property does not have the expected value. Property: {0}", propertyName);
         }
 
-        private static void AssertRepositoryIsValid(JarInfo.Repository repository)
+        private static void AssertExpectedRulesExist(DiagnosticAnalyzer analyzer, string actualRuleFilePath)
         {
-            Assert.IsNotNull(repository, "Repository should not be null");
-
-            AssertPropertyHasValue(repository.Key, "Repository Key");
-            AssertPropertyHasValue(repository.Name, "Repository Name");
+            //XDocument rulesXml = XDocument.Load(actualRuleFilePath);
```

Comment:
TODO: once I've picked up @dbolkensteyn 's changes to the metadata classes, it should be possible to load the rules file into a strongly-typed RulesModel.